### ISSUE TITLE
Add state machine code to testing utilities

### ISSUE THREAD

**Issue by JoelBender** [JoelBender/bacpypes#11]
There is a state machine implementation for testing sitting in an untracked project.  Bring it in.

**Comment by JoelBender** [comment]
The code is now in `tests/state_machine.py` and there are corrisponding tests in the `tests/test_utilities/test_state_machine.py` modules.

In order to make sure that the timeout() state transitions are working correctly, I wrote a *TimeMachine* class (in the `tests/time_machine.py` module) which overrides the *TaskManager* which simulates the passage of time.  The combinations of these classes are designed to build tests like "send a Who-Is, wait 3 seconds and make sure an I-Am is not returned."  I had to change a few things in `bacpypes/task.py` to make this work, so it will need some additional testing before I merge it into the trunk.
